src/app/shared/thank-you/thank-you.component.ts

```diff
@@ -6,16 +8,11 @@ import { Component, OnInit } from '@angular/core';
   styleUrls: ['./thank-you.component.scss']
 })
 export class ThankYouComponent implements OnInit {
+  thankUser!: User;
 
-  constructor() { }
+  constructor(private usersService: UsersService) { }
 
   ngOnInit(): void {
-  }
-
-  thankUser = {
-    fullName: 'Carl Jhonson',
-    name: 'Carl',
-    lastName: 'Jhonson',
-    image: '',
+    this.thankUser = this.usersService.getUsers().sort((a, b) => b.xp - a.xp)[0];
```

Comment:
if you will have empty array of users and will try to get 0 element of this array - you will get an error. please resolve this

Reply:
I resolve it by adding *ngIf="thankUser" in template